api/v1beta1/awscluster_webhook.go

```diff
@@ -115,6 +115,12 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 					r.Spec.ControlPlaneLoadBalancer.Name, "field is immutable"),
 			)
 		}
+		if existingLoadBalancer.HealthCheckProtocol != nil && existingLoadBalancer.HealthCheckProtocol.String() != newLoadBalancer.HealthCheckProtocol.String() {
```

Comment:
`existingLoadBalancer = nil` means SSL is being used. Should block setting it to TCP.

For nil case, the only acceptable value is SSL.

Reply:
Unit tests should cover these scenarios.

Reply:
Done